src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseParser.cs

```diff
@@ -187,7 +187,14 @@ public static unsafe bool GetResponseHeader(
                 index = originalIndex;
 
                 buffer = new char[bufferLength];
-                return GetResponseHeader(requestHandle, infoLevel, ref buffer, ref index, out headerValue);
+                fixed (char* pBuffer = &buffer[0])
+                {
+                    if (QueryHeaders(requestHandle, infoLevel, pBuffer, ref bufferLength, ref index))
```

Comment:
Shouldn't we update `lastError` if this fails?

Reply:
Fixed.